flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/FlinkService.java

```diff
@@ -172,10 +174,14 @@ public boolean isJobManagerPortReady(Configuration config) {
                     clusterClient.cancel(jobID).get(1, TimeUnit.MINUTES);
                     break;
                 case SAVEPOINT:
+                    final String savepointDirectory =
+                            Preconditions.checkNotNull(
+                                    conf.get(CheckpointingOptions.SAVEPOINT_DIRECTORY));
+                    final long timeout = operatorConfiguration.getFlinkClientTimeout().getSeconds();
```

Comment:
The flink client timeout that we have in the operator config is mostly calibrated for listing actions and has 10 sec default. I think that might cause problems here

Reply:
I will keep the timeout as before and introduce a new config option if necessary in the future.

Reply:
How about taking the checkpoint timeout from the flink config directly here? (And use the default for that from flink)

Reply:
I am afraid we could not get it from the flink config and it could also be overridden by user.

```
env.getCheckpointConfig().setCheckpointTimeout(60000);
```

Reply:
It should somehow be configurable I agree on that otherwise this will fail for larger states :/

Reply:
Worst case we could introduce a savepoint timeout config for now

Reply:
Is the [execution.checkpointing.timeout](https://nightlies.apache.org/flink/flink-docs-master/zh/docs/deployment/config/#execution-checkpointing-timeout) what we want?

Reply:
The user configured timeout has high priority. Right?

Reply:
We could ignore that in this case and make a note of it

Reply:
I agree with using the execution.checkpointing.timeout

Reply:
I will verify whether it also take effect for savepoint, not only checkpoint.